ikarus/finiteelements/mechanics/loads/volume.hh

```diff
@@ -0,0 +1,146 @@
+      = std::nullopt) const -> ScalarType {
+        const auto uFunction = displacementFunction(par, dx);
+        ScalarType energy    = 0.0;
+        const auto& lambda   = par.getParameter(Ikarus::FEParameter::loadfactor);
+        for (const auto& [gpIndex, gp] : uFunction.viewOverIntegrationPoints()) {
+            const auto uVal                              = uFunction.evaluate(gpIndex);
+            Eigen::Vector<double, Traits::worlddim> fext = volumeLoad(geo_->global(gp.position()), lambda);
+            energy -= uVal.dot(fext) * geo_->integrationElement(gp.position()) * gp.weight();
+        }
+      }
+
+    /**
+     * @brief Applies the distributed volume loads on the element domain.
+     *
+     * @tparam ScalarType The scalar type for the force vector.
+     * @param par The FERequirementType object.
+     * @param force The force vector to be updated.
+     * @param dx Optional displacement vector.
+     */
+    template <typename ScalarType>
+    void calculateVector(const FERequirementType& par, typename Traits::template VectorType<ScalarType> force,
+                             const std::optional<const Eigen::VectorX<ScalarType>>& dx = std::nullopt) const {
+      using namespace Dune::DerivativeDirections;
+      using namespace Dune;
+      const auto uFunction = displacementFunction(par, dx);
+      const auto geo       = this->localView().element().geometry();
+      const auto& lambda   = par.getParameter(Ikarus::FEParameter::loadfactor);
+      for (const auto& [gpIndex, gp] : uFunction.viewOverIntegrationPoints()) {
+        const Eigen::Vector<double, worldDim> fext = volumeLoad(geo.global(gp.position()), lambda);
+        const double intElement                    = geo.integrationElement(gp.position()) * gp.weight();
+        for (size_t i = 0; i < numberOfNodes; ++i) {
+          const auto udCi = uFunction.evaluateDerivative(gpIndex, wrt(coeff(i)));
+          force.template segment<worldDim>(worldDim * i) -= udCi * fext * intElement;
+        }
+      }
+    }
+
+      /**
+       * @brief Calculates the matrix stiffness for the given FERequirementType.
+       *
+       * @param par The FERequirementType object.
+       * @param K Matrix to store the calculated stiffness.
+       */
+      template <typename ScalarType>
+      void calculateMatrix(const FERequirementType& par, typename Traits::template MatrixType<> K,
+                           const std::optional<const Eigen::VectorX<ScalarType>>& dx = std::nullopt) const {
+          DUNE_THROW(Dune::NotImplemented, "calculateMatrix is not implemented for the class Volume");
+      }
+
+      bool isApplied(){return isApplied_;}
+
+  private:
+    std::shared_ptr<const Geometry> geo_;
+    Dune::CachedLocalBasis<std::remove_cvref_t<LocalBasisType>> localBasis;
+    std::function<Eigen::Vector<double, worldDim>(const Dune::FieldVector<double, worldDim>&, const double&)>
```

Comment:
Only the `volumeLoad` and a pointer to the displacementbased fe should be sufficient hopefully. Also similar in the constructor